src/include/processor/operator/partitioner.h

```diff
@@ -20,7 +20,13 @@ struct PartitionerFunctions {
 // partitioning methods. For example, copy of rel tables require partitioning on both FWD and BWD
 // direction. Each partitioning method corresponds to a PartitioningState.
 struct PartitioningBuffer {
-    std::vector<std::unique_ptr<common::DataChunkCollection>> partitions;
+    struct NodeGroup {
```

Comment:
I'd rather not use the term `NodeGroup` here, just not to be confused with the one in `storage/store/node_group.h`. You can consider using `Partition`.

You can also wrap `std::vector<std::unique_ptr<storage::ColumnChunk>>` as `ColumnChunkCollection`.